geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateDataSourceCommandDUnitTest.java

```diff
@@ -99,6 +102,46 @@ public void testCreateDataSource() throws Exception {
     verifyThatNonExistentClassCausesGfshToError();
   }
 
+  @Test
+  public void testCreateDataSourceWithJarOptionDoesNotThrowDriverError() {
+    String URL = "jdbc:mysql://localhost/";
+    IgnoredException.addIgnoredException(
+        "No suitable driver");
+    IgnoredException.addIgnoredException(
+        "create data-source failed");
+
+    // aquire the jar to be used
+    final String jdbcJarName = "mysql-connector-java-8.0.15.jar";
+    final String jdbcDriverClassName = "com.mysql.cj.jdbc.Driver";
+    File mySqlDriverFile = loadTestResource("/" + jdbcJarName);
+    assertThat(mySqlDriverFile).exists();
+    String jarFile = mySqlDriverFile.getAbsolutePath();
+
+    // attempt to create the data source without a deployed jar or a --driver-jar
+    gfsh.executeAndAssertThat(
+        "create data-source --name=mySqlDataSource --username=mySqlUser --password=mySqlPass --pooled=false --url=\""
+            + URL + "\"")
+        .statusIsError();
+
+    gfsh.executeAndAssertThat("deploy --jar=" + jarFile).statusIsSuccess();
+
+    IgnoredException.removeAllExpectedExceptions();
+    IgnoredException.addIgnoredException(
+        "create data-source failed");
+    IgnoredException.addIgnoredException(
+        "com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure");
+    IgnoredException.addIgnoredException(
+        "Access denied for user 'mySqlUser'@'localhost'");
+    IgnoredException.addIgnoredException(
+        "Failed to connect to \"mySqlDataSource\"");
+
+    gfsh.executeAndAssertThat(
+        "create data-source --name=mySqlDataSource --username=mySqlUser --password=mySqlPass --pooled=false --url=\""
+            + URL + "\" --jdbc-driver-class=" + jdbcDriverClassName)
+        .statusIsError().doesNotContainOutput("No suitable driver");
+
```

Comment:
This comment really pertains to the `CreateDataSourceCommand` itself, but consider that if this command fails the `jndi` xml definition will not be persisted - is that really what you want in all failing situations?

Reply:
This is the intended behavior of the command at the moment. Addressing potential issues with this is something we can do at a later date, as it's not directly tied to the current PR.